05/ContactForm.js

```diff
@@ -1,10 +1,159 @@
-import React from 'react';
+import React, { useReducer, useState } from 'react';
 
-import account from './account';
-
-const ContactForm = () => {
-    console.log(account);
-    return <form>??</form>;
+const initialState = {
+    name: '',
+    email: '',
+    phone: '',
+    subject: '',
+    message: '',
 };
 
+function reducer(state, action) {
+    switch (action.type) {
+    case 'UPDATE_FIELD':
+        return {
+            ...state,
+            [action.field]: action.value,
+        };
+    case 'RESET_FORM':
+        return initialState;
+    default:
+        return state;
+    }
+}
+
+function ContactForm() {
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const [errors, setErrors] = useState({});
+    const validate = () => {
+        const newErrors = {};
+        if (!state.name) newErrors.name = 'Imię i nazwisko jest wymagane.';
+        if (!state.email) newErrors.email = 'Adres e-mail jest wymagany.';
+        if (!state.subject) newErrors.subject = 'Temat jest wymagany.';
+        if (!state.message) newErrors.message = 'Wiadomość jest wymagana.';
+        setErrors(newErrors);
+        return Object.keys(newErrors).length === 0;
```

Comment:
Można zrobić bardziej uniwersalnie, np.: https://devmentor.pl/b/tworzenie-i-walidacja-formularza-javascript
... i do tego autoamtycznie generować pola formularza - to byłby czad!!! :D -> https://devmentor.pl/b/dynamiczne-renderowanie-komponentow-w-react